### ISSUE TITLE
GNOME Bluetooth settings break

### ISSUE THREAD

**Issue by heftig** [bus1/dbus-broker#15]
When the system broker is in use, the Bluetooth panel in the GNOME Control Center is unusable. Also, it seems Bluetooth isn't enabled with the session even if it should be. The Shell can still control Bluetooth, though.

I see the following message in the journal: `gnome-bluetooth-panel.desktop[17923]: Agent registration failed: GDBus.Error:org.freedesktop.DBus.Error.AccessDenied: Sender is not authorized to send message`

`obexd` also repeatedly complains about `AccessDenied`.

dbus-broker v1
GNOME 3.24
Arch Linux x86_64

**Comment by teg** [comment]
Thanks for testing!

This is an known issue and something we need to come up with a solution for (we have several options).

This is due to bluez using at_console policies, see https://github.com/bus1/dbus-broker/wiki/Deviations.

dbus-broker should warn about this in the logs, do you see that? Suggestions for making the log messages clearer would be welcome.

A local fix would be to copy bluez' policy file and change the at_console policy to say at_console=false. The implications would be that apps running under a non-logged-in uid could access bluez. This is probably not a problem, but something one should be aware of.

**Comment by heftig** [comment]
> dbus-broker should warn about this in the logs, do you see that?

Here are messages that sound like warnings:

```
dbus-broker-launch[469]: Reply/Error policy in /usr/share/dbus-1/system.conf +56: Explicit policies on replies and errors are deprecated and ignored
dbus-broker-launch[469]: Reply/Error policy in /usr/share/dbus-1/system.conf +57: Explicit policies on replies and errors are deprecated and ignored
dbus-broker-launch[469]: Reply/Error policy in /usr/share/dbus-1/system.conf +61: Explicit policies on replies and errors are deprecated and ignored
dbus-broker-launch[469]: Reply/Error policy in /usr/share/dbus-1/system.conf +62: Explicit policies on replies and errors are deprecated and ignored
dbus-broker-launch[469]: Policy record in console-context in /etc/dbus-1/system.d/com.redhat.NewPrinterNotification.conf +6: at_console=true is deprecated and ignored
dbus-broker-launch[469]: Policy record in console-context in /etc/dbus-1/system.d/com.redhat.PrinterDriversInstaller.conf +6: at_console=true is deprecated and ignored
dbus-broker-launch[469]: Policy record in console-context in /etc/dbus-1/system.d/bluetooth.conf +28: at_console=true is deprecated and ignored
dbus-broker-launch[469]: Falling back to resolving auxillary groups using nss, this is racy and may cause deadlocks. Update to a kernel with SO_PEERGROUPS support.
```

> Suggestions for making the log messages clearer would be welcome.

Use levels (priorities). It looks like all messages from the broker are currently at INFO level (6).

**Comment by heftig** [comment]
I'm not sure if this is related, but I had problems with NetworkManager that were resolved after disabling the system broker. The most visible issue was `nmcli con up id foo` not exiting after activating the connection (perhaps it doesn't see the replies or the signals it expects).

**Comment by dvdhrm** [comment]
As an FYI: We rework the policy logging right now. Once that is pushed, we will be able to better diagnose this. So we are still looking into this!

**Comment by Keruspe** [comment]
Has an issue been opened for upstream gnome-bluetooth?

**Comment by dvdhrm** [comment]
@heftig, the `nmcli` issue (and the `vpn` issue) should be fixed via #55 (at least it fixes both issues on my machine). If it is still broken for you, please re-open this, or comment on it (I will leave this bug open, anyway, until I submitted a patch to BlueZ).

**Comment by dvdhrm** [comment]
The bluetooth situation is fixed with -git. See #56 for detailed background information. The implementation is still subject to change, but the idea behind our solution will very likely stay.

I will close this now. If there is still an issue, please re-open or file separately!

Thanks a lot for the report and analysis!
